src/api/currency/index.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * src/api/internal/index.js
+ *
+ * Endpoint do servidor principal com os módulos conectores das blockchains
+ */
+
+const currencies = require('./currencies')
+const common = require('./common')
+
+class CurrencyApi {
+	constructor() {
+		/**
+		 * @description Faz com que os módulos de cryptocurrencies sejam
+		 * acessíveis pela API
+		 */
+		this.currencies = currencies
+		/**
+		 * @description Insere propriedades/funções da common no módulo
+		 * 'this.currencies[currency]' sem sobrescrever propriedades que tenham
+		 * o mesmo nome em um módulo individual. Ou seja, se uma função existir
+		 * na common e no módulo de uma currency, a que irá prevalecer será
+		 * o do módulo da currency
+		 */
+		Object.keys(this.currencies).forEach(currency => {
+			this.currencies[currency] = {...common, ...this.currencies[currency]}
+		})
+	}
+}
+
+module.exports = singleton = new CurrencyApi()
+
+require('./listener')(singleton)
```

Comment:
nao compreendo duas coisas nessa linha: 
1-por que a utima linha do arquivo é um require e por que ele nao pode ser feito antes
2- singleton é uma palavra reservada de js ou vc so usou ela pois vc quer que essa classe seja uma instancia unica e resolveu especificar com o nome ou algo do genero

Reply:
1 - o listener precisa ter acesso as funções da classe declarada no index. A maneira escolhida pra fazer isso foi passar a referência da classe como argumento do listener, e isso só pode acontecer quando a classe for instanciada, que só acontece na linha 30
2 - [Singleton](https://pt.wikipedia.org/wiki/Singleton) é um padrão pra garantir que apenas uma única instância da classe exista no programa inteiro, que é exatamente o que essa classe precisa. Eu usei o nome 'singleton' pra deixar isso evidente, do tipo 'essa classe precisa ser um singleton', mas não é palavra reservada ou algo assim, só o nome da variável mesmo